.github/workflows/build_deps.yml

```diff
@@ -26,55 +17,60 @@ on:
 jobs:
   build_deps:
     name: Build Deps
-    if: ${{ !cancelled() && (inputs.build-deps-only || inputs.force-build || inputs.valid-cache != true) }}
     runs-on: ${{ inputs.os }}
+    outputs:
+      artifact-name: ${{ env.ARTIFACT_NAME }}
+      artifact-path: ${{ env.DEPS_PATH }}
     env:
-      date:
+      DO_BUILD: ${{ inputs.build-deps-only || inputs.force-build }}
+      DEPS_PATH: ${{ github.workspace }}/deps/build${{ inputs.os != 'macos-14' && '/OrcaSlicer_dep' || '' }}
+      ARTIFACT_NAME: OrcaSlicer_dep_${{ inputs.os }}_${{ inputs.arch }}
     steps:
-
-      # Setup the environment
       - name: Checkout
         uses: actions/checkout@v6
         with:
           lfs: 'true'
 
       - name: load cached deps
-        uses: actions/cache@v4
+        uses: actions/cache/restore@v4
+        id: cache-load
         with:
-          path: ${{ inputs.cache-path }}
-          key: ${{ inputs.cache-key }}
+          path: ${{ env.DEPS_PATH }}
+          key: ${{ inputs.os }}-${{ inputs.arch }}-cache-orcaslicer_deps-build-${{ hashFiles('deps/**') }}
+          restore-keys: ${{ inputs.os }}-${{ inputs.arch }}-cache-orcaslicer_deps-build
```

Comment:
I think this might cause issues that are hard to track down? 
It will silently build Orca with outdated dependencies, which will be very difficult to debug.  
It’s better to make it fail immediately.

Reply:
It's why further lines all use: 

`        if: ${{ !cancelled() && (env.DO_BUILD == 'true' || steps.cache-load.outputs.cache-hit != 'true') && inputs.os == 'windows-latest' }}`

`cache-hit` is only true when the exact key matches; if a "restore key" is used, it is false. 

And though I do force cmake to run with force-build, I'm not forcing it to do a *clean* build, which is why I put it in in the first place, to ensure it can download everything. So I should change that.

Reply:
Removed, it's now building XOR cache (at least that's the intention...)